jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanReader.java

```diff
@@ -155,6 +158,22 @@ public Object read(JSONReader r, JsonParser p) throws IOException
                 return _constructors.create(p.getLongValue());
             case START_OBJECT:
                 {
+                    // [jackson-jr#148] Record deser support (2.18)
+                    if (_isRecordType) {
+                        final List<Object> values = new ArrayList<>();
+
+                        String propName;
+                        for (; (propName = p.nextFieldName()) != null;) {
+                            BeanPropertyReader prop = findProperty(propName);
+                            if (prop == null) {
+                                handleUnknown(r, p, propName);
+                                continue;
+                            }
+                            values.add(prop.getReader().readNext(r, p));
```

Comment:
This is wrong: it cannot just blindly append entries -- needs to figure which index it should go to.

I should have caught this in code review.

Reply:
Fixed in https://github.com/FasterXML/jackson-jr/pull/163